src/main/java/com/worksap/fig/lang/Seq.java

```diff
@@ -404,4 +431,189 @@ default int findLastIndex(Predicate<T> condition) {
+    }
+
+    /**
+     * Remove elements which satisfy the condition at the original seq.
+     *
+     * @return The seq itself after the change
+     */
+    default Seq<T> removeIf$(BiPredicate<T, Integer> condition) {
+        Objects.requireNonNull(condition);
+        final Iterator<T> each = iterator();
+        int index = 0;
+        while (each.hasNext()) {
+            if (condition.test(each.next(), index)) {
+                each.remove();
+            }
+            index++;
+        }
+        return this;
+    }
+
+    /**
+     * Get elements which satisfy the condition, resulting a new seq without changing the original one.
+     *
+     * @return The new seq
+     */
+    default Seq<T> getIf(Predicate<T> condition) {
+        Objects.requireNonNull(condition);
+        Seq<T> copy = new SeqImpl<>();
+        this.forEach(e -> {
+            if (condition.test(e))
+                copy.add(e);
+        });
+        return copy;
+    }
+
+    /**
+     * Get elements which satisfy the condition, resulting a new seq without changing the original one.
+     *
+     * @return The new seq
+     */
+    default Seq<T> getIf(BiPredicate<T, Integer> condition) {
+        Objects.requireNonNull(condition);
+        Seq<T> copy = new SeqImpl<>();
+        this.forEachWithIndex((e, i) -> {
+            if (condition.test(e, i))
+                copy.add(e);
+        });
+        return copy;
+    }
+
+    /**
+     * Get the element with index. If the index is illegal, return a default value
+     * @param i index
+     * @param defaultValue default value
+     * @return the element or default value
+     */
+    default T get(int i, T defaultValue) {
+        if (i < 0 || i >= this.size()) {
+            return defaultValue;
+        }
+        return this.get(i);
+    }
+
+    /**
+     * Returns a new {@link Seq} built by concatenating the times copies of self.
+     * @return the new seq
+     */
+    default Seq<T> repeat(int times) {
+        if (times < 0)
+            throw new IllegalArgumentException("times must be a positive number.");
+        Seq<T> result = new SeqImpl<>();
+        while (times > 0) {
+            result.addAll(this);
+            times--;
+        }
+        return result;
+    }
+    /**
+     * Change the original {@link Seq} by concatenating the times copies of self.
+     * @return the seq itself after the change
+     */
+    default Seq<T> repeat$(int times) {
+        if (times < 0)
+            throw new IllegalArgumentException("times must be a positive number.");
+        else if (times == 0)
+            clear();
+        else if (times == 1)
+            ;
+        else {
+            times--;
+            while (times > 0) {
+                this.addAll(this);
+                times--;
+            }
+        }
+        return this;
+    }
+
+    /**
+     * Returns the element at index. A negative index counts from the end of self.
+     */
+    default T at(int index) {
```

Comment:
How about overriding `get` method to achieve the same function?

Which do you think better?